### ISSUE TITLE
Erroneously reported reverse-range-loop 

### ISSUE THREAD

**Issue by iirelu** [Manishearth/rust-clippy#409]
```rust
error: this range is empty so this for loop will never run [-D reverse-range-loop]
 for x in -50i32..50 {
```

This works fine (x goes -50, -49, etc) but clippy complains about it for the wrong reason.

**Comment by birkenfeld** [comment]
Thanks for the report!
